kikuchipy/signals/ebsd.py

```diff
@@ -1004,6 +1005,275 @@ def dictionary_indexing(
             get_orientation_similarity_map=get_orientation_similarity_map,
         )
 
+    def orientation_refinement(
+        self,
+        xmap: CrystalMap,
+        master_pattern,
+        detector: EBSDDetector,
+        energy: Union[int, float],
+        mask: Optional[np.ndarray] = None,
+        method: Optional[str] = None,
+        method_kwargs: Optional[dict] = None,
+        trust_region: Optional[list] = None,
+        compute: bool = True,
+    ) -> CrystalMap:
+        """Performs an orientation refinement using the initial indexing
+        results stored in a single phase
+        :class:`~orix.crystal_map.CrystalMap` and the fixed
+        detector-sample geometry. The refinement uses a subset of the
+        optimization routines implemented in scipy and attempts to
+        maximize the similarity between experimental and simulated
+        patterns as calculated by the NCC.
+
+        Parameters
+        ----------
+        xmap : CrystalMap
+            A crystal map storing the results of the initial EBSD
```

Comment:
Should note that the sample reference frame used is TSL's (can take this explanation from the dictionary_indexing() docstring).